base/src/com/thoughtworks/go/util/FilenameUtil.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.util;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+public class FilenameUtil {
+    public static boolean isSubdirectoryOf(String parent, String subdirectory) throws IOException {
```

Comment:
Did you forget to remove the old impl? Also, this method name is ambiguous, IMO. I don't know if there are cases where this answer requires us to actually resolve file paths — can we confirm?

Reply:
Didn't forget to remove it, its been left behind intentionally. There are some very specific cases of this validation being done wrt server only directories and similar stuff for agent-only directories which need file system level resolution. Comment about method renaming makes sense, will do.